File: group2-fintech/starter-workspace/scripts/db_ping.py
#!/usr/bin/env python3
"""Prove the workshop database is reachable, before you build anything against it.

    python scripts/db_ping.py

This is toolchain, not application code — it makes no assumption about your
schema and creates nothing.
"""
import os
import re
import sys

URL = os.environ.get(
    "DATABASE_URL", "postgresql://swiftkyc:swiftkyc@localhost:5433/swiftkyc"
)
SAFE = re.sub(r":[^:@/]*@", ":****@", URL)


def main() -> int:
    try:
        import psycopg
    except ImportError:
        print("psycopg is not installed. Run:  pip install -r requirements.txt")
        return 1

    try:
        with psycopg.connect(URL, connect_timeout=5) as conn:
            with conn.cursor() as cur:
                cur.execute(
                    "select current_database(), version(), (now() at time zone 'UTC')"
                )
                row = cur.fetchone()
        if row is None:
            print(f"connected to {SAFE} but the server returned nothing — that is not normal")
            return 1
        db, version, utc_now = row
    except Exception as e:
        print(f"\ncannot reach the database at {SAFE}")
        print(f"  {e}\n")
        print("Start it first, from this folder:")
        print("    docker compose up -d")
        print('`docker compose ps` should show "healthy" before you retry.')
        print("If Docker Desktop itself is not running, start that first.")
        return 1

    print(f"  url      {SAFE}")
    print(f"  database {db}")
    print(f"  server   {version.split(',')[0]}")
    print(f"  utc now  {utc_now.isoformat()}")
    print("\ndatabase is up.")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: group1-retail/starter-workspace/scripts/db-ping.mjs
#!/usr/bin/env node
// Proves the workshop database is reachable, before you build anything against it.
//
//     npm run db:ping
//
// This is toolchain, not application code — it makes no assumption about your
// schema and creates nothing.
import pg from 'pg';

const url =
  process.env.DATABASE_URL ?? 'postgres://pointhub:pointhub@localhost:5432/pointhub';
const safe = url.replace(/:[^:@/]*@/, ':****@');
const client = new pg.Client({ connectionString: url, connectionTimeoutMillis: 5000 });

try {
  await client.connect();
  const { rows } = await client.query(
    "select current_database() as db, version() as v, now() at time zone 'UTC' as utc_now",
  );
  console.log(`  url      ${safe}`);
  console.log(`  database ${rows[0].db}`);
  console.log(`  server   ${rows[0].v.split(',')[0]}`);
  console.log(`  utc now  ${rows[0].utc_now.toISOString()}`);
  console.log('\ndatabase is up.');
} catch (err) {
  // pg leaves `message` empty for some connection failures — fall back to the code.
  const why = err.message?.trim() || err.code || String(err);
  console.error(`\ncannot reach the database at ${safe}`);
  console.error(`  ${why}\n`);
  console.error('Start it first, from this folder:');
  console.error('    docker compose up -d');
  console.error('`docker compose ps` should show "healthy" before you retry.');
  console.error('If Docker Desktop itself is not running, start that first.');
  process.exitCode = 1;
} finally {
  await client.end().catch(() => {});
}
